internal/k8s/controllers/config_controller_test.go

```diff
@@ -0,0 +1,335 @@
+						AutoAssign:    pointer.BoolPtr(false),
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "pool2",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.IPAddressPoolSpec{
+						Addresses: []string{
+							"30.0.0.0/8",
+						},
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "pool3",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.IPAddressPoolSpec{
+						Addresses: []string{
+							"40.0.0.0/25",
+							"40.0.0.150-40.0.0.200",
+							"40.0.0.210 - 40.0.0.240",
+							"40.0.0.250 - 40.0.0.250",
+						},
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "pool4",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.IPAddressPoolSpec{
+						Addresses: []string{
+							"2001:db8::/64",
+						},
+					},
+				},
+			},
+			BGPAdvs: []v1beta1.BGPAdvertisement{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "adv1",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.BGPAdvertisementSpec{
+						AggregationLength: pointer.Int32Ptr(32),
+						LocalPref:         uint32(100),
+						Communities:       []string{ /* TODO CRD Add communities"bar", */ "1234:2345"},
+						IPAddressPools:    []string{"pool1"},
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "adv2",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.BGPAdvertisementSpec{
+						AggregationLength:   pointer.Int32Ptr(24),
+						AggregationLengthV6: pointer.Int32Ptr(64),
+						IPAddressPools:      []string{"pool1"},
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "adv3",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.BGPAdvertisementSpec{
+						IPAddressPools: []string{"pool2"},
+					},
+				},
+			},
+			L2Advs: []v1beta1.L2Advertisement{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "l2adv1",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.L2AdvertisementSpec{
+						IPAddressPools: []string{"pool3"},
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "l2adv2",
+						Namespace: testNamespace,
+					},
+				},
+			},
+			PasswordSecrets: map[string]corev1.Secret{
+				"bgpsecret": {Type: corev1.SecretTypeBasicAuth, ObjectMeta: v1.ObjectMeta{Name: "bgpsecret", Namespace: testNamespace},
+					Data: map[string][]byte{"password": []byte([]byte("nopass"))}},
+			},
+			LegacyAddressPools: []v1beta1.AddressPool{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "legacypool1",
+						Namespace: testNamespace,
+					},
+					Spec: v1beta1.AddressPoolSpec{
+						Addresses: []string{
+							"10.21.0.0/16",
+							"10.51.0.0/24",
+						},
+						Protocol:      "bgp",
+						AvoidBuggyIPs: true,
+						AutoAssign:    pointer.BoolPtr(false),
+					},
+				},
+			},
+		}
+
+		calledForceReload = false
+		mockForceReload   = func() { calledForceReload = true }
+	)
+
+	if err := v1beta1.AddToScheme(scheme); err != nil {
```

Comment:
This is is distracting. Wrap everything in a "newFakeClient" function that you put in the bottom of the file

Reply:
done